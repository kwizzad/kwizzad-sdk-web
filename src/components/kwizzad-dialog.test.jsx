/* globals it */
/* globals describe */
import React from 'react';
import renderer from 'react-test-renderer';
import KwizzadDialog from '../components/kwizzad-dialog.jsx';

describe('KwizzadDialog', () => {
  it('renders without problems', () => {
    const props = {
      placement: { requestAd: () => {}, dismissAd: () => {} },
      placementId: 'myPlacement',
      apiKey: 'abc123',
    };
    const element = <KwizzadDialog {...props} />;
    const tree = renderer.create(element).toJSON();

    // See https://facebook.github.io/jest/docs/en/snapshot-testing.html#content to learn how this
    // works.
    expect(tree).toMatchSnapshot();
  });
});
