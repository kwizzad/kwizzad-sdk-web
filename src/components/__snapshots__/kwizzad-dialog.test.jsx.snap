// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KwizzadDialog renders without problems 1`] = `
<div
  className="kwizzad-modal-dialog kwizzad-modal-dialog-iframe kwizzad-kwizzad-modal-dialog-hidden"
>
  <div
    className="kwizzad-modal-dialog-fullscreen-overlay"
    onClick={[Function]}
  />
  <div
    className="kwizzad-modal-dialog-inner kwizzad-"
  >
    <div
      className="kwizzad-modal-dialog-content"
    >
      <button
        className="kwizzad-close-dialog"
        onClick={[Function]}
      >
        <svg
          height="0.5em"
          version="1.1"
          viewBox="168 231 31 31"
          width="0.5em"
        >
          <polygon
            fill="#FFFFFF"
            fillRule="evenodd"
            id="Ã—"
            points="180.121094   246.582031 168.90625 235.296875 172.351562 231.816406     183.601562 243.066406   194.957031 231.816406 198.4375 235.191406   187.046875   246.582031 198.367188   257.902344 194.957031 261.277344   183.601562 250.027344   172.351562 261.207031   168.976562 257.832031"
            stroke="none"
          />
        </svg>
      </button>
      <iframe
        allowFullScreen={true}
        className="kwizzad-iframe"
        frameBorder="0"
        marginHeight="0"
        marginWidth="0"
        src="undefined&openTarget=_blank"
      />
    </div>
  </div>
</div>
`;
